Lightning Flows Module

martademenyosf opened the pull request:
creating the screen flow and the corresponding classes

Commit:
Lightning Flows Module

creating the screen flow and the corresponding classes

Commit:
Lightning Flows Module - add custom setting to read Weather page data

aliozdemirosf commented on force-app/main/default/classes/OSF_UpdateAccountsBatch.cls:
```diff
@@ -0,0 +1,91 @@
+    // instance member to retain state across transactions
+    public Integer recordsProcessed = 0;
+
+    private static final String WEATHER_IMG_URL = 'https://openweathermap.org/img/wn/';
```
please move it to custom setting or metadata

martademenyosf replied:
I deleted this variable to don't read again from custom setting. The weather info data are returned using the OSF_WeatherReader.getWeatherInfo method instead of OSF_WeatherReader.executeAPICall

Commit:
Lightning flow Module

- delete the WEATHER_IMG_URL from OSF_UpdateAccountsBatch class to don't read again from custom setting
- add test class for OSF_UpdateAccountsBatch
- in OSF_UpdateWeatherInfoForAccountsTest  class insert the custom setting in the setup method

aliozdemirosf commented on force-app/main/default/classes/OSF_UpdateWeatherInfoForAccountsTest.cls:
```diff
@@ -0,0 +1,65 @@
+
+        Weather_Http_List_Setting__c settings = new Weather_Http_List_Setting__c();
+
+        settings.Name = 'WeatherInfo';
+        settings.Weather_API_URL__c = 'https://api.openweathermap.org/data/2.5/weather';
+        settings.Weather_IMG_URL__c = 'https://openweathermap.org/img/wn/';
+        settings.APPID__c = '9ab4d1340bf4d964ff4412a9068f7edc';
+        System.debug('WeatherInfo: ' + settings);
```
even on the testing, you should never expose any sensitive data such as api keys or credentials

aliozdemirosf commented on force-app/main/default/classes/OSF_UpdateAccountsBatch.cls:
```diff
@@ -0,0 +1,85 @@
+
+        for (Account acc: source) {
+             if (acc.BillingCity != null) {
+                List<OSF_WeatherReaderResult> results = OSF_WeatherReader.getWeatherInfo(new List<String>{acc.BillingCity});
```
Question: what is the use case and you sent list of billing cities when you are only having a single value per record?

martademenyosf replied:
I have to sant a list because this method is called from the flow, where the returned and the parameter type should be a list: https://developer.salesforce.com/docs/atlas.en-us.apexcode.meta/apexcode/apex_classes_annotation_InvocableMethod.htm

aliozdemirosf commented on force-app/main/default/classes/OSF_UpdateAccountsBatchTest.cls:
```diff
@@ -0,0 +1,58 @@
+        settings.Name = 'WeatherInfo';
+        settings.Weather_API_URL__c = 'https://api.openweathermap.org/data/2.5/weather';
+        settings.Weather_IMG_URL__c = 'https://openweathermap.org/img/wn/';
+        settings.APPID__c = '9ab4d1340bf4d964ff4412a9068f7edc';
```
even on the testing, you should never expose any sensitive data such as api keys or credentials

aliozdemirosf commented on force-app/main/default/classes/OSF_WeatherReader.cls:
```diff
@@ -0,0 +1,92 @@
+        WEATHER_URL = settings.Weather_API_URL__c;
+        APPID = settings.APPID__c;
+        WEATHER_IMG_URL = settings.Weather_IMG_URL__c;
+        System.debug('WEATHER_URL: ' + WEATHER_URL);
+        System.debug('WEATHER_IMG_URL: ' + WEATHER_IMG_URL);
+        System.debug('APPID: ' + APPID);  
```
please clean up debugs from all the classes

martademenyosf replied:
done

Commit:
Lightning Flows Module - changes after code review

- delete all system.debug
- add mock value to APPID__c
